browser/ephemeral_storage/ephemeral_storage_tab_helper.cc

```diff
@@ -104,9 +109,18 @@ void EphemeralStorageTabHelper::CreateEphemeralStorageAreasForDomainAndURL(
 
   std::string session_partition_id = StringToSessionStorageId(
       content::GetSessionStorageNamespaceId(web_contents()),
-      "/ephemeral-session-storage");
-  session_storage_namespace_ =
-      content::CreateSessionStorageNamespace(partition, session_partition_id);
+      kSessionStorageSuffix);
+
+  auto* rfh = web_contents()->GetOpener();
+  session_storage_namespace_ = content::CreateSessionStorageNamespace(
+      partition, session_partition_id,
+      // clone the namespace if there is an opener
+      // https://html.spec.whatwg.org/multipage/browsers.html#copy-session-storage
```

Comment:
Will this clone copy non-ephemeral session storage data into the ephemeral session storage? I'm pretty sure that's why I didn't do this in the past.

Reply:
it shouldn't be, I'm only passing the id of the parent session storage and it passes all the tests on https://dev-pages.brave.software/storage/ephemeral-storage.html

Reply:
and since the opener will be suppressed if it's not same domain the owner window will always have ephemeral storage enabled, but it might be worth adding a check in case there's some edge case with subdomains

Reply:
looks like it should be ok
```// If the namespace doesn't exist or it's not populated yet, just create
      // an empty session storage by not marking it as pending a clone.
      if (clone_from_ns == namespaces_.end() ||
          !clone_from_ns->second->IsPopulated()) {
        break;
      }```

Reply:
would be nice to have a browser test